Setup check and added test examples

AnishaSrivastava1411 opened the pull request:
Added jibx-maven-plugins and dependencies for setup.

goneall commented on .gitignore:
```diff
@@ -0,0 +1 @@
+/target/
```
I would add the .idea, .classpath, .project, .settings, and .ant-target-build.xml to the gitignore since these are IDE specific.  When we complete the project, it should be able to run just using the maven commands from the command lines.  @rtgdk let me know if you disagree.

rtgdk replied:
+1

goneall commented on pom.xml:
```diff
@@ -0,0 +1,148 @@
+            <execution>
+             <configuration>
+              <schemaLocation>http://www.jibx.org/sampleschema/home/anisha/Documents <!-- Remember the trailing '/' --></schemaLocation>
+            
```
This should be a directory local to the project (e.g. a src/main/config)  You can put the schema in the local directory.  That way it will get checked into the repo and you will not be referencing files on your local system.

AnishaSrivastava1411 commented:
@goneall , @rtgdk  please review. I have done changes.

rtgdk commented on .classpath:
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
@AnishaSrivastava1411 Remove these unnecessary files by adding them to .gitignore

AnishaSrivastava1411 commented:
I have made requested changes. Please review @rtgdk @goneall

rtgdk commented on target/maven-archiver/pom.properties:
```diff
@@ -0,0 +1,5 @@
+#Generated by Maven
```
You need to add `target` also in gitignore. Basically every file that is specific to your development environment should not be commited.

AnishaSrivastava1411 commented:
> @AnishaSrivastava1411 Can you also add different codes you tried for loading the file? Basically, you can commit whatever code you try out, this way it would be better to keep track of deliverable and easier to check history.

@rtgdk  Sure, I will do it.

goneall commented:
I see the Example1.java file in the target directory and everything looks like it is now working.

In terms of having the target folder in the gitignore, it is a good practice to have all generated files ignored by git and they should be generated by the users.

Since this project is focused on the generated source files, it would be nice to see those in the pull requests.  Perhaps we can change folder used for the generated java files.  After the directory is created, you can update the pom.xml configuration for the schema-gen goal.  That way, you could ignore the target directory which includes other generated stuff and leave the generated java files in git.

AnishaSrivastava1411 commented:
@goneall  @rtgdk please review.

1.  I have changed the directory for generated source codes to **src/main/source-code** and put the target directory back to .gitignore. 

2.  I took spdx-xml-ref.xsd file directly from spdx-xsd repo and converted into java classes.

3. I have also updated README.md

But, there exists a problem too. At a time only one .xsd file's converted codes remain in the folder ie, in src/main/source-code. My earlier generated java class for example1.xsd got deleted when I tried spdx-xml-ref.xsd and vice-versa.

goneall commented:
>  At a time only one .xsd file's converted codes remain in the folder ie, in src/main/source-code. My earlier generated java class for example1.xsd got deleted when I tried spdx-xml-ref.xsd and vice-versa.

I believe only the classes for the XSD files configured in the plugin should be generated in the source directory.  I would expect the others to be removed.  If I understand the description, it sounds like the plugin is working as expected.

You should be able to add multiple .xsd files if you want to generate multiple class files.

goneall commented:
Could you use the XSD file from xmlschema branch of the spdx-spec git repository: https://github.com/spdx/spdx-spec/blob/xmlschema/schemas/spdx-schema.xsd  That will be the official schema.  If there are any issues with the schema itself, let me know.

AnishaSrivastava1411 commented:
> > At a time only one .xsd file's converted codes remain in the folder ie, in src/main/source-code. My earlier generated java class for example1.xsd got deleted when I tried spdx-xml-ref.xsd and vice-versa.
> 
> I believe only the classes for the XSD files configured in the plugin should be generated in the source directory. I would expect the others to be removed. If I understand the description, it sounds like the plugin is working as expected.
> 
> You should be able to add multiple .xsd files if you want to generate multiple class files.

I did try adding multiple files and it worked well. All the attached files were converted into java classes.

AnishaSrivastava1411 commented:
@goneall @rtgdk I have tried mild customisation to generated classes. 

1. This removes snippets of schema present as comments in .java files. Attribute used: **--show-schema="false"** from terminal.
2. Also removes annotations if present in .java files. Attribute used: **--delete-annotations** from terminal.

It seems like, we might require to edit schemas to customise files so that number of classes generated is reduced.
Do we need to look ahead in reducing the number of files by inter-relating them?  
I also wanted to ask, what more changes I should try in generated files.

goneall commented:
> It seems like, we might require to edit schemas to customise files so that number of classes generated is reduced.

You should replace the schema with the file from the SPDX spec: https://github.com/spdx/spdx-spec/blob/xmlschema/schemas/spdx-schema.xsd

That should produce a much more manageable number of classes.

goneall commented:
> I also wanted to ask, what more changes I should try in generated files.

Once we have this PR updated with the target removed and the schema replaced, we should have a better idea what to customize.

Some of the customizations will require [Codgen Extensions](http://jibx.sourceforge.net/fromschema/codegen-extends.html).  Setting this up looks a bit complex, so I would suggest creating a very simple extensions that makes a minor modification to the code generation and get the infrastructure working.  Once that is in place, you can systematically customize the code generation for the java classes starting with one of the simpler classes like Checksum.

AnishaSrivastava1411 commented:
Sir, I updated the `src/main/config` folder, it has schema: [ https://github.com/spdx/spdx-spec/blob/xmlschema/schemas/spdx-schema.xsd]
I also updated pom.xml to change the default `target/generated-sources` to 'src/main/source-code' .
@goneall @rtgdk Please review.

goneall commented:
@AnishaSrivastava1411 Looks good - It looks like a manageable number of classes now.  We can also review the generated source in the PR's and all of the other build artifacts looks like they are ignored.

I'll go ahead a merge this so that we can start working on the customization and be able to review them as new PR's.

goneall commented:
@rtgdk I just noticed you have some requested changes - I think they are already made, but I wanted to check before merging.  Let me know if this is OK to merge.

rtgdk commented:
@goneall  LGTM. Approved. You can merge.
